File: packages/contracts/python/veyra_contracts/events.py
"""Event contracts. docs/architecture/12-EVENTS.md"""

from __future__ import annotations

from datetime import UTC, datetime
from typing import Any
from uuid import uuid4

from pydantic import BaseModel, Field

from veyra_contracts.enums import EventType


class Event(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid4()))
    type: EventType
    payload: dict[str, Any] = Field(default_factory=dict)
    correlation_id: str
    timestamp: datetime = Field(default_factory=lambda: datetime.now(UTC))

File: packages/contracts/typescript/src/events.ts
// Mirrors packages/contracts/python/veyra_contracts/events.py's Event
// model — the exact JSON shape broadcast over the real `/events`
// WebSocket (services/local-api/app/api/events.py).

import type { AgentState, EventType, VisemeShape } from "./enums";

export interface VisemeFrame {
  shape: VisemeShape;
  start_ms: number;
  duration_ms: number;
}

// docs/phase-6/AVATAR-ARCHITECTURE.md — the payload shape for
// `voice.ui_state.changed`. `visemes`/`outcome` are only ever present
// alongside `agent_state: "SPEAKING"` (see VoiceConversationManager).
export interface AvatarUIStatePayload {
  agent_state: AgentState;
  visemes?: VisemeFrame[];
  outcome?: AgentState;
}

export interface VeyraEvent {
  id: string;
  type: EventType;
  payload: Record<string, unknown>;
  correlation_id: string;
  timestamp: string;
}
